### DISCUSSION TITLE
Is a USB transmitter enough?

### DISCUSSION THREAD

**Discussion by RainerHeintzmann** [tbnobody/OpenDTU#1233] [Q&A] [answered]
Can the same effect not also be achieved by using a USB-based transceiver?
Something like this?
https://www.ebay.de/itm/152699998304?mkcid=16&mkevt=1&mkrid=707-127634-2357-0&ssspo=SfYuxmzcQym&sssrc=2047675&ssuid=&widget_ver=artemis&media=COPY

A USB-based device would make everything so much easier, as one would not need to build anything and could directly connect it to the laptop or the Fritz!Box. Or am I missing something important here?

**Comment by stefan123t** [comment, marked as answer]
@RainerHeintzmann we started out with an NRF24L01+ (mind the plus, signifying 250MBps mode capability!) connected to the GPIOs of a Raspberry Pi.

While you could theoretically use it in polling mode this tends to take considerable amount of CPU cycles, hence we switched to ESP32 and other microcontrollers to receive the packets signalled by the IRQ pin of the NRF24L01+.

I kind of doubt that the vendor has many specs for this module or the IRQ ist somehow handed over USB ? 

Besides that who would be willing to write the driver to collect the data and for which OS ? 
You name the Fritz!Box, which has indeed a USB port but not many are developing firmware for it using the freetz sources available, let alone a Linux kernel driver for this module.

For the newer HMS/HMT modules you will need a different radio chip anyway. Maybe you simply consider to buy an OpenDTU Fusion v2.1 module which comes ready to run with both radios.

**Reply by RainerHeintzmann** [reply]
Thanks! I guess the argument was to stay with as little as possible additional devices. But if this model(s) are anyway not really sold any longer, and if their protocol is way off common standards, and you need direct access to some interrupts, than it is really not worth the effort.

**Comment by Solarteur** [comment]
An additional argument for the existing solution: 

With my reception conditions, I can only receive the signal from the inverter in the garage or in the garden - with additional WLAN it works throughout the house.
